sources/packages/api/src/route-controllers/education-program-offering/education-program-offering.controller.ts

```diff
@@ -226,7 +226,26 @@ export class EducationProgramOfferingController {
 
   @AllowAuthorizedParty(AuthorizedParties.student)
   @Get("location/:locationId/education-program/:programId/options-list")
-  async getProgramOfferingsForLocation(
+  async getProgramOfferingsByLocation(
+    @Param("locationId") locationId: number,
+    @Param("programId") programId: number,
+  ): Promise<OptionItem[]> {
+    const offerings =
+      await this.programOfferingService.getProgramOfferingsForLocation(
+        locationId,
+        programId,
+      );
+
+    return offerings.map((offering) => ({
+      id: offering.id,
+      description: offering.name,
+    }));
+  }
+
+  @AllowAuthorizedParty(AuthorizedParties.institution)
+  @HasLocationAccess("locationId")
+  @Get("location/:locationId/education-program/:programId/offerings-list")
+  async getProgramOfferingsForLocationForInstitution(
```

Comment:
This is what is causing 1 of the 2 code smells. We are duplicating intentionally because this "HasLocationAccess". Can you please confirm?

Reply:
The short answer is yes. We need a method to return a list of programs (the same for offerings) and this method is consumed by students and institutions.
Students will have access to all the programs/offerings while institutions are going to have access only to their respective programs and offerings.
So, we have two methods with different authorization. To avoid impact the returned payload size, we are also keeping the data returned so far as only a key/value. Besides that, I would imagine these methods returning different data for students and institutions, which, besides only authorization, would justify the two methods.
Always open to suggestions 😉

Reply:
Thats exactly what I thought. At this point I don't see an alternative. You can disregard these 2 codesmells